app/questionnaire/path_finder.py

```diff
@@ -212,12 +245,15 @@ def get_next_location(self, current_location):
 
             return routing_path[current_location_index + 1]
 
+        routing_path = self.build_path(current_section)
```

Comment:
What's the significance of rebuilding the path by calling `build_path` here that's different to the path returned when calling `routing_path` on line 227?

Reply:
routing_path will return an already built path if it exists and the answer store isn't dirty. If we make it to this point we need to re-evaluate the path, as it is no longer valid (for instance, when skipping a question)

Reply:
Sorry, I can't see what happens between calling `routing_path` and `build_path` that would make the path any different.